### ISSUE TITLE
Caching runs

### ISSUE THREAD

**Issue by Shivansh-007** [Shivansh-007/piston-cli#55]
Sometimes you want to run code as it was during the last run or due to internet failure or piston being down you couldn't execute. To support these three cases and ease of running the same data as the last run, piston should cache run data in `~/.cache/piston-cli` (accordingly for other OS). This cache would be the hash of that time. On every run it would hash the time and save a file containing data something like:
```json
{
  "language": "python",
  "stdin": "\n join string of stdin",
  "args": ["list of args"],
  "code": "code to be runned in here"
}
```

If #53 gets implemented before this gets implemented then include the other options too.

### Working with cache through piston cli

- You can specify the filename of the cache file, say `130334460710706012`, then you can run it like this `piston --cache 130334460710706012`. And it would directly send the data to the piston v1/2 endpoint.
- You can also specify the files using `-1`, `-2`, etc. The `-1` means to run the last cache file, the `-2` means to run the second last cache file and so on.
- To clear the cache one can do `--clear-cache`, which would then ask for confirmation and clear the cache.
- One can also configure the location of these cache files in `~/.config/piston/config.yml`

**Comment by Shivansh-007** [comment]
I will be working on this.

**Comment by Shivansh-007** [comment]
After thinking more about this, I am not agreeing with this issue. If anyone still wants, we can have a discussion here.

**Comment by Shivansh-007** [comment]
Implemented the bad connection caching in #73
